contracts/BatchAuction.sol

```diff
@@ -1,13 +1,23 @@
 pragma solidity ^0.5.0;
 
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
-contract BatchAuction {
+
+contract BatchAuction is Ownable {
 
     uint16 public constant MAX_ACCOUNT_ID = 100;
+    uint8 public constant MAX_TOKENS = 30;
 
     mapping (address => uint16) public publicKeyToAccountMap;
     mapping (uint16 => address) public accountToPublicKeyMap;
 
+    uint8 public numTokens;
+    mapping (address => uint8) public tokenAddresToIdMap;
+    mapping (uint8 => address) public tokenIdToAddressMap;
+
+    event RegisteredToken(address tokenAddress, uint8 tokenId);
```

Comment:
Do we really need this Event? I think the dB doesn’t have to track tokens since they are stored in the smart contract without any compression. 

So any participant could just query the smart contract directly for this information. No need to store it twice.

Reply:
Sounds fair to me! I will remove this!